src/Aspire.Hosting.AWS/Lambda/LambdaLifecycleHook.cs

```diff
@@ -22,10 +21,51 @@ internal class LambdaLifecycleHook(ILogger<LambdaEmulatorResource> logger, IProc
 
     public async Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
     {
+        var classLibraryProjectPaths = 
+            appModel.Resources
+                .OfType<LambdaProjectResource>()
+                .Where(x =>
+                {
+                    if (!x.TryGetLastAnnotation<LambdaFunctionAnnotation>(out var functionAnnotation))
+                        return false;
+
+                    if (!functionAnnotation.Handler.Contains("::"))
+                        return false;
+
+                    return true;
+                })
+                .ToList();
+        
         LambdaEmulatorAnnotation? emulatorAnnotation = null;
         if (appModel.Resources.FirstOrDefault(x => x.TryGetLastAnnotation<LambdaEmulatorAnnotation>(out emulatorAnnotation)) != null && emulatorAnnotation != null)
         {
             await ApplyLambdaEmulatorAnnotationAsync(emulatorAnnotation, cancellationToken);
+
+            foreach (var projectResource in classLibraryProjectPaths)
+            {
+                var projectMetadata = projectResource.Annotations
+                    .OfType<IProjectMetadata>()
+                    .First();
+                var lambdaFunctionAnnotation = projectResource.Annotations
+                    .OfType<LambdaFunctionAnnotation>()
+                    .First();
+            
+                var installPath = await GetCurrentInstallPathAsync(cancellationToken);
```

Comment:
Should you add a check if `installPath` is empty string and if so log an error and continue? Actually looking at the utility methods `targetFramework` and `assemblyName` should have the check as well.